### ISSUE TITLE
Project area discrepency question

### ISSUE THREAD

**Issue by MattTheCuber** [mikedh/trimesh#2227]
So I recently found that trimesh can compute projected area. This is great, as I previously had to use shapely and I don't really like that method. I tried it out for the z plane and it worked perfect, but I can't seem to get the same numbers when using the other planes.

### Shapely method
```python
mesh = trimesh.load_mesh("/path/to/stl")

triangles_x = [shapely.Polygon(mesh.vertices[tri, 1:]) for tri in mesh.faces]
merged_x = shapely.union_all(triangles_x)
triangles_y = [shapely.Polygon(mesh.vertices[tri][:, [0, 2]]) for tri in mesh.faces]
merged_y = shapely.union_all(triangles_y)
triangles_z = [shapely.Polygon(mesh.vertices[tri, :2]) for tri in mesh.faces]
merged_z = shapely.union_all(triangles_z)

print(merged_x.area, merged_y.area, merged_z.area)
```
Prints `136.43230002926032 320.87002785291054 395.49557873527743`

### Trimesh method
```python
mesh = trimesh.load_mesh("/path/to/stl")

projected_x = mesh.projected([1, 0, 0])
projected_y = mesh.projected([0, 1, 0])
projected_z = mesh.projected([0, 0, 1])

print(projected_x.area, projected_y.area, projected_z.area)
```
Prints `55.790675139305506 26.725690822297 395.03629678814224`

Note: my mesh is a irregular shape, kind of resembling a rectangular prism, and the shapely values match what I would expect.

**Comment by mikedh** [comment]
Yeah, I've noticed mediocre results of `mesh.projected` for non-simple shapes too. Using an even lightly cleaned up `unary_union` option does do a nicer job than the current trimesh method, albeit roughly 3x slower in my tests (this is on a subdivided 73856 face mesh):
```
unary_union: 2.7377541065216064
mesh.projected: 1.1850848197937012
```

It's close enough speed-wise to think about removing the region-logic in trimesh and just unioning the triangles. It also seems like `unary_union` is way more stable than I remember. 

I thought there ought to be a better algorithm that uses the 2D triangles and adjacency information from the mesh to do the unions in one pass over the adjacency but I couldn't quite get that to work just now.  

```
import time

import numpy as np
from shapely.geometry import Polygon
from shapely.ops import unary_union

import trimesh

if __name__ == "__main__":
    tic = [time.time()]

    m = trimesh.load("models/Duck.glb", force="mesh", merge_tex=True, merge_norm=True)

    normal = [0, 0, 1]

    # project to 2D
    v = m.vertices[:, :2]
    # only take normal-positive faces
    faces = m.faces[np.dot(m.face_normals, normal) > 0.0]

    # small numeric tolerance
    eps = 1e-10

    # do a simple unary union of triangles
    r = (
        unary_union([Polygon(f) for f in v[np.column_stack((faces, faces[:, :1]))]])
        .buffer(eps)
        .buffer(-eps)
    )

    tic.append(time.time())

    # use the trimesh method
    p = m.projected(normal=normal)
    
    tic.append(time.time())

    print("unary_union: {}\nmesh.projected: {}".format(*np.diff(tic)))

```

**Comment by mikedh** [comment]
Not really a good answer to this, but I did add a `mesh.projected(..., precise=True)` option which will just union the 2D triangles in shapely, and probably mostly answers this question. I still kind of think there ought to be a good way to use the 3D connectivity information and DFS it doing a single "point in polygon" test each time... but I guess that is pretty much `unary_union` with extra steps. If someone has a good algorithm for this (i.e.: `f(2.5D triangle soup, connectity) -> polygon`) PR's would be super welcome!

**Comment by MattTheCuber** [comment]
Upon further research I have discovered that Clipper uses the [Vatti Clipping Algorithm](https://en.m.wikipedia.org/wiki/Vatti_clipping_algorithm). There are several more algorithm described here https://en.m.wikipedia.org/wiki/Boolean_operations_on_polygons. However, Vatti seems to be the best method (due to highest tolerance while still being fast). There are probably many other methods, but these came up in my research. I am curious now what method Shapely uses and what method `trimesh.projected(precise=False)` uses...
